kebechet/managers/thoth/thoth.py

```diff
@@ -0,0 +1,141 @@
+# as a subprocess as pipenv keeps path to the virtual environment in the global context that is not
+# updated on subsequent calls.
+
+
+class ThothManager(ManagerBase):
+    """Manage updates of dependencies."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize update manager."""
+        self._repo = None
+        # We do API calls once for merge requests and we cache them for later use.
+        self._cached_merge_requests = None
+        super().__init__(*args, **kwargs)
+
+    @property
+    def repo(self):
+        """Get repository on which we work on."""
+        return self._repo
+
+    @repo.setter
+    def repo(self, repo: git.Repo):
+        """Set repository information and all derived information needed."""
+        self._repo = repo
+        self.slug = repo.remote().url.split(':', maxsplit=1)[1][:-len('.git')]
+
+    @property
+    def sha(self):
+        """Get SHA of the current head commit."""
+        return self.repo.head.commit.hexsha
+
+    # TODO: determine how branch name will be constructed
+    def _construct_branch_name(self) -> str:
+        """Construct branch name for the updated dependency."""
+        return f"{_BRANCH_NAME}-{self.sha}"
+
+    
+    def _git_push(self, commit_msg: str, branch_name: str, files: list, force_push: bool = False) -> None:
+        """Perform git push after adding files and giving a commit message."""
+        #  self.repo.git.checkout('HEAD', b=branch_name)
+        self.repo.index.add(files)
+        self.repo.index.commit(commit_msg)
+        self.repo.remote().push(branch_name, force=force_push)
+
+    
+    def _open_merge_request(self, branch_name: str, labels: list, files: list) -> typing.Optional[int]:
+        """Open a pull/merge request for dependency update."""
+        commit_msg = "Auto generated update"
+        body = "Pipfile.lock updated by kebechet-thoth manager"
+
+        # Delete branch if it didn't change Pipfile.lock
+        diff = self.repo.git.diff('master', files)
+        if diff == "":
+            return
+        
+        # push force always to keep branch up2date with the recent master and avoid merge conflicts.
+        self._git_push(":pushpin: " + commit_msg, branch_name, files, force_push=True)
+        
+        for mr in self._cached_merge_requests:
+            if mr.head_branch_name == branch_name:
+                return
+        
+        merge_request = self.sm.open_merge_request(commit_msg, branch_name, body, labels)
+        return merge_request
+
+    @staticmethod
+    def _advise_wrap():
+        with open("Pipfile", "r") as pipfile, open("Pipfile.lock", "a+") as piplock:
+            # TODO: take into account config values for recommendation type,
+            #       limit latest versions, no static analysis, no wait,
+            #       limit, count, debug
+            res = lib.advise(pipfile.read(), piplock.read())
```

Comment:
Maybe it would worth to move this to thamos repository and provide something like `lib.advise_on_repo()` which would operate on root of the git repository so we can use the same logic everywhere. WDYT?

Reply:
This sounds like a good idea to me, I think it's best to just keep the same default values for everything else.